netlify/netlify-plugin-gatsby
# feat: support disabling of Netlify function generation

Currently we have no way of knowing at build time whether there are any DSG or SSR pages, or Gatsby Functions. For this reason we always have to generate the handler functions. This PR adds the ability to disable these both manually by setting an env var, and programatically by creating a file in the cache dir. The latter feature is to allow `gatsby-plugin-netlify` to disable it automatically (PR to follow).

To test:

1. Run `NETLIFY_SKIP_GATSBY_FUNCTIONS=1 ntl build` and observe the message in the `onBuild` handler.
 2. Change the build command to `gatsby build && touch .cache/.nf-skip-gatsby-functions` then run `ntl build` and observe a different message.

<img width="1150" alt="CleanShot 2022-01-26 at 15 59 15@2x" src="https://user-images.githubusercontent.com/11183523/151167163-6c4fd094-f120-4105-adc3-ad7e5f916771.png">


Fixes #166. Fixes #219. Fixes #214
